### PR TITLE
Add download solution button and localization

### PR DESCRIPTION
**Pull request by orronai**

### CODE REVIEW

File: lms/lmsdb/models.py [hunk 1]
```diff
@@ -8,6 +8,7 @@
     Type, Union, cast,
 )
 
+from flask_babel import gettext
```
Comment (yammesicka) [inline lms/lmsdb/models.py:11]: We should import it as `_` to make it less noisy (it's a known convention)

Look at example of greater projects, such as [Apache's Incubator Superset](https://github.com/apache/incubator-superset/blob/a6101f72c96d89e3fbf9999f16cf3b5147f73d8f/superset/utils/core.py).

File: lms/lmstests/public/flake8/services.py [hunk 2]
```diff
@@ -92,10 +93,12 @@ def _fire_notification_if_needed(self):
 
         errors_len = len(self._errors)
         exercise_name = self.solution.exercise.subject
-        msg = (
-            f'הבודק האוטומטי נתן {errors_len} '
-            f'הערות על תרגילך "{exercise_name}".'
-        )
+        msg = ''.join((
+            gettext('הבודק האוטומטי נתן'),
```
Comment (yammesicka) [inline lms/lmstests/public/flake8/services.py:97]: The break here will make it harder to translate things when it come to other grammars.
Make it a single string and remove the fstrings.

The same applies for the rest of the project.

File: lms/lmsweb/views.py [hunk 3]
```diff
@@ -40,6 +42,18 @@
 HIGH_ROLES = {str(RoleOptions.STAFF), str(RoleOptions.ADMINISTRATOR)}
 MAX_REQUEST_SIZE = 2_000_000  # 2MB (in bytes)
 
+DIRECTION = 'rtl' if LOCALE == 'he' else 'ltr'
```
Comment (yammesicka) [inline lms/lmsweb/views.py:45]: We should have somewhere a `set()` of rtl languages and `DIRECTION` should be set to `rtl` if `LOCATE in RTL_LANGUAGES`

File: lms/lmsweb/views.py [hunk 4]
```diff
@@ -111,7 +125,11 @@ def login():
             return fail(400, "The URL isn't safe.")
         return redirect(next_url or url_for('main'))
 
-    return render_template('login.html')
+    return render_template(
+        'login.html',
+        direction=DIRECTION,
+        locale=get_locale(),
```
Comment (yammesicka) [inline lms/lmsweb/views.py:131]: Isn't there a better way to localize themes?

File: lms/static/grader.js [hunk 5]
```diff
@@ -200,6 +200,15 @@ function addNewCommentButtons(elements) {
 }
 
 
+const lang = document.querySelector('html').lang;
+let addCommentString;
```
Comment (yammesicka) [inline lms/static/grader.js:204]: We must find a better solution for the JS, this is not a production ready code.

File: lms/static/my.css [hunk 6]
```diff
@@ -12,19 +21,19 @@ div#code-view {
   text-align: left;
 }
 
-::-webkit-input-placeholder {
+:lang(he) ::-webkit-input-placeholder {
```
Comment (yammesicka) [inline lms/static/my.css:24]: I prefer not to use the `:lang` pseudo class. It forcefully chooses the language for the user.

Use regular class like `.rtl-language` and put the conditions whether or not to append this class in the HTML

**Review by yammesicka — CHANGES_REQUESTED** [review]

**Review by yammesicka — CHANGES_REQUESTED** [review]

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: babel.cfg [hunk 7]
```diff
@@ -0,0 +1,2 @@
+[python: **.py]
+[jinja2: **/templates/**.html]
```
Comment (yammesicka) [inline babel.cfg:2]: Are you sure the file is needed?

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: babel.cfg (hunk 7 above)
Comment (orronai) [reply babel.cfg:2]: The templates dir isn't really needed. But we must have the html files. Otherwise we won't get the translations of them via Babel

**Review by yammesicka — APPROVED** [review]

**Pull request merged**